### PR TITLE
[terra-form-select] Combobox custom placeholder remove

### PR DESCRIPTION
**Pull request by ashishkcerner**
### Summary
To ensure the Form - Select Combobox meets accessibility standards for 2.4.6 Heading and Labels we need to ensure the place holder text does not have the ability to have custom text. String for what should display as placeholder text in the Acceptance Criteria

**What was changed:**
- The option to have custom placeholder text is removed
-  The string "-Select-" displays as the placeholder text.

**Why it was changed:**
Combobox shouldn't have the option for custom placeholder according to design doc

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [x] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9731 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-form-select/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+*Fixed
+  *Fixed the `combobox` won't take custom placeholder text and only string `-Select- ` will be shown.
```
Comment (supreethmr) [inline packages/terra-form-select/CHANGELOG.md:6]:
```suggestion
* Changed
  * Removed support for custom placeholder on the `combobox`
```

File: packages/terra-form-select/src/Combobox.jsx [hunk 2]
```diff
@@ -88,10 +88,6 @@ const propTypes = {
    * Callback function invoked for each option on search change. function(searchValue, option)
    */
   optionFilter: PropTypes.func,
-  /**
-   * Placeholder text.
-   */
-  placeholder: PropTypes.string,
```
Comment (supreethmr) [inline packages/terra-form-select/src/Combobox.jsx:94]: this is breaking change !! we need to update the upgrade guide and change log entry with title `Breaking Change`

File: packages/terra-form-select/CHANGELOG.md (hunk 1 above)
Comment (KV106606Viswanath) [reply packages/terra-form-select/CHANGELOG.md]: Updated with title "Breaking change" as per below comment,

File: packages/terra-form-select/src/Combobox.jsx (hunk 2 above)
Comment (KV106606Viswanath) [reply packages/terra-form-select/src/Combobox.jsx:94]: Updated

**Comment by supreethmr** [comment]
This pull request is closed for the reasons stated in Jira:UXPLATFORM-9731

**Pull request closed without merging**